File: 2016/1.py
steps = []
with open("input_1.txt") as f:
    lines = f.readlines()
    for l in lines:
        ins = l.split("\n")[0].split(", ")

position = 0 + 0j
pos = [position]
direction = 0 + 1j
for i in ins:
    if i[0] == 'L':
        direction *= 1j
    else:
        direction *= -1j

    steps = int(i[1:])
    for _ in range(steps):
        position += direction
        if position not in pos:
            pos.append(position)
        else:
            res = (int(abs(position.real) + abs(position.imag)))
            print ("🎄Part 2: {}".format(res))
            import sys
            sys.exit()
res = (int(abs(position.real) + abs(position.imag)))
print ("🎄Part 1: {}".format(res))

File: 2016/1.jl
include("utils.jl")
using .AoCUtils

function move(ins)
    pos::Complex = 0+0im
    direction::Complex = 0+1im
    records = []
    record = true
    
    for i in ins
        steps = parse(Int, i[2:length(i)])
        if i[1] == 'L'
            direction *= 1im
        else
            direction *= -1im
        end

        for s in 1:steps
            pos += direction
            if pos in records
                println("🎄Part 2: $(abs(real(pos)) + abs(imag(pos)))")
                record = false
            else
                if record
                    append!(records, pos)
                end
            end
        end
    end
    println("🎄Part 1: $(abs(real(pos)) + abs(imag(pos)))")
end

line = readSingleLine("input_1.txt")
move(split(line, ", "))

